src/parser/name_resolution.c

```diff
@@ -5074,22 +5113,20 @@ pt_remake_dblink_select_list (PARSER_CONTEXT * parser, PT_SPEC_INFO * class_spec
       PT_NODE *id_node = parser_new_node (parser, PT_NAME);
       if (id_node == NULL)
 	{
-	  return ER_FAILED;
+	  error = ER_DBLINK;
+	  goto error_exit;
 	}
 
       id_node->info.name.original = "c";
       if ((dblink_table->cols = pt_mk_attr_def_node (parser, id_node, NULL)) == NULL)
 	{
-	  return ER_FAILED;
+	  error = ER_DBLINK;
+	  goto error_exit;
```

Comment:
It seems necessary to restore parser->custom_print before returning at line 5127.
An easy way to do this is to 'goto error_exit' instead of 'return NO_ERROR'. 
In this case, 'clean_up' or something similar can be a better name instead of the label 'error_exit' .

Reply:
Ok, good point.